external/@wordpress/blocks/index.js

```diff
@@ -19,14 +28,20 @@ export function registerBlock( namespace, block, settings ) {
  * @return {?Object}           Block settings
  */
 export function getBlockSettings( namespace, block ) {
+	if ( ! blocks[ namespace ] || ! blocks[ namespace ][ block ] ) {
+		return null;
+	}
 
+	return blocks[ namespace ][ block ];
 }
 
 /**
  * Returns all registered blocks.
  *
- * @return {Object} Block settings keyed by block name
+ * @return {Array} Blocks settings
```

Comment:
Yeah, I'm worried we'll find the namespacing will get in our way more than it helps. I'm not totally convinced about it. It's nice to have the consistency with similar APIs for the REST API, but I wonder if we could just allow a simple name and perhaps encourage a convention around including some namespace in the name itself: like `"wp/text"` (or without namespace for "core" types).